src/lib/Libutil/misc_utils.c

```diff
@@ -347,9 +347,13 @@ pbs_strcpy(char *dest, const char *src)
  */
 char *pbs_strncpy(char *dest, const char *src, size_t n)
 {
-    strncpy(dest, src, n - 1);
-    dest[n - 1] = '\0';
-    return dest;
+	if (strlen(src) < n - 1)
+		strcpy(dest, src);
+	else {
+		strncpy(dest, src, n - 1);
```

Comment:
strncpy will do another strlen(src), whereas you have done this already in line 350. You could store it in a register variable first time and use memcpy instead of strncpy in the second case (353)

Reply:
I don't think you meant to store the value of strlen in a register because if the length of src is already more than space destination has then we can just do memcpy to the size of the destination. Maybe, the implementation you are referring to involves doing strnlen and using only memcpy.
I'll change strncpy to memcpy anyways.

Reply:
ok